security-analysis/src/main/java/com/powsybl/security/Security.java

```diff
@@ -148,17 +176,99 @@ public static String printLimitsViolations(Network network) {
+     */
+    @Deprecated
     public static String printLimitsViolations(List<LimitViolation> violations) {
         return printLimitsViolations(violations, LimitViolationFilter.load());
     }
 
+    public static String printLimitsViolations(List<LimitViolation> violations, Network network, LimitViolationFilter filter) {
+        return printLimitsViolations(violations, network, filter, TableFormatterConfig.load());
+    }
+
+    public static String printLimitsViolations(List<LimitViolation> violations, Network network, LimitViolationFilter filter, TableFormatterConfig formatterConfig) {
+        Objects.requireNonNull(violations);
+        Objects.requireNonNull(filter);
+        Objects.requireNonNull(formatterConfig);
+
+        TableFormatterFactory formatterFactory = new AsciiTableFormatterFactory();
+        Writer writer = new StringWriter();
+        List<LimitViolation> filteredViolations = filter.apply(violations, network);
+
+        NumberFormat numberFormat = getFormatter(formatterConfig.getLocale(), 4, 4);
+        NumberFormat percentageFormat = getFormatter(formatterConfig.getLocale(), 2, 2);
+
+        try (TableFormatter formatter = formatterFactory.create(writer,
+                "",
+                formatterConfig,
+                new Column(COUNTRY),
+                new Column(BASE_VOLTAGE)
+                    .setHorizontalAlignment(HorizontalAlignment.RIGHT),
+                new Column(EQUIPMENT + " (" + filteredViolations.size() + ")"),
+                new Column(END),
+                new Column(VIOLATION_TYPE),
+                new Column(VIOLATION_NAME),
+                new Column(VALUE)
+                    .setHorizontalAlignment(HorizontalAlignment.RIGHT)
+                    .setNumberFormat(numberFormat),
+                new Column(LIMIT)
+                    .setHorizontalAlignment(HorizontalAlignment.RIGHT)
+                    .setNumberFormat(numberFormat),
+                new Column(ABS_VALUE_LIMIT)
+                    .setHorizontalAlignment(HorizontalAlignment.RIGHT)
+                    .setNumberFormat(numberFormat),
+                new Column(LOADING_RATE)
+                    .setHorizontalAlignment(HorizontalAlignment.RIGHT)
+                    .setNumberFormat(percentageFormat))) {
+            filteredViolations.stream()
+                    .sorted(Comparator.comparing(LimitViolation::getSubjectId))
+                    .forEach(writeLineLimitsViolations(network, formatter));
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        return writer.toString().trim();
+    }
+
+    private static Consumer<? super LimitViolation> writeLineLimitsViolations(Network network, TableFormatter formatter) {
+        return violation -> {
+            try {
+                formatter.writeCell(LimitViolation.getCountry(violation, network) != null ? LimitViolation.getCountry(violation, network).name() : "")
```

Comment:
The return of LimitViolation.getCountry(violation, network) cannot be null

Reply:
done